src/main/java/com/telefonica/jwk/Entrypoint.java

```diff
@@ -0,0 +1,113 @@
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.KeyUse;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.cli.PosixParser;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
+import java.security.Security;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class Entrypoint {
+  private static String DEFAULT_KEY_SIZE = "2048";
+  private static Options options;
+
+  public static void main(String[] args) {
+
+    Security.addProvider(new BouncyCastleProvider());
+
+    options = new Options();
+
+    options.addOption("s", true, "Key Size in bits, required for RSA and oct key types. Must be an integer divisible by 8");
+    options.addOption("u", true, "Usage, one of: enc, sig (optional)");
+    options.addOption("S", false, "Wrap the generated key in a KeySet");
+
+    CommandLineParser parser = new PosixParser();
+    try {
+      CommandLine cmd = parser.parse(options, args);
+
+      String size = cmd.getOptionValue("s");
+      String use = cmd.getOptionValue("u");
+      boolean keySet = cmd.hasOption("S");
+      Algorithm algorithm = JWEAlgorithm.RSA_OAEP_256;
+
+      KeyUse keyUse = null;
+      if (use != null) {
+        if (use.equals("sig")) {
+          keyUse = KeyUse.SIGNATURE;
+          algorithm = JWSAlgorithm.RS256;
+        } else if (use.equals("enc")) {
+          keyUse = KeyUse.ENCRYPTION;
+        } else {
+          printUsageAndExit("Invalid key usage, must be 'sig' or 'enc', got " + use);
+        }
+      }
+
+
+      // surrounding try/catch catches numberformatexception from this
+      if (Objects.isNull(size)) {
+        size = DEFAULT_KEY_SIZE;
+      }
+
+      Integer keySize = Integer.decode(size);
+      if (keySize % 8 != 0) {
+        printUsageAndExit("Key size (in bits) must be divisible by 8, got " + keySize);
+      }
+
+      JWK jwk = RSAKeyMaker.make(keySize, keyUse, algorithm);
+
+      // round trip it through GSON to get a prettyprinter
+      Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+      printKey(keySet, jwk, gson);
+    } catch (NumberFormatException e) {
+      printUsageAndExit("Invalid key size: " + e.getMessage());
+    } catch (ParseException e) {
+      printUsageAndExit("Failed to parse arguments: " + e.getMessage());
+    }
+  }
+
+
+  private static void printKey(boolean keySet, JWK jwk, Gson gson) {
+    if (keySet) {
+      JWKSet jwkSet = new JWKSet(jwk);
+      JsonElement json = new JsonParser().parse(jwkSet.toJSONObject(false).toJSONString());
+      System.out.println(gson.toJson(json));
+    } else {
+      JsonElement json = new JsonParser().parse(jwk.toJSONString());
+      System.out.println(gson.toJson(json));
```

Comment:
```suggestion
      System.out.println(GSON.toJson(json));
```
but is it not better to create a new file with the new keys?

Reply:
We want to parse the output, support for file may be used in future ...